packages/governance/src/ballotBuilder.js

```diff
@@ -0,0 +1,43 @@
+// @ts-check
+
+import { assert, details as X } from '@agoric/assert';
+
+// CHOOSE_N: voter indicates up to N they find acceptable (N might be 1).
+// ORDER: voter lists their choices from most to least favorite.
+// WEIGHT: voter lists their choices, each with a numerical weight. High
+//   numbers are most preferred.
+
+const ChoiceMethod = {
+  CHOOSE_N: 'choose_n',
+  ORDER: 'order',
+  WEIGHT: 'weight',
+};
+
+const buildBallot = (method, question, positions, maxChoices = 0) => {
+  const choose = (...chosenPositions) => {
+    assert(
+      chosenPositions.length <= maxChoices,
+      X`only ${maxChoices} position(s) allowed`,
+    );
+
+    for (const position of chosenPositions) {
+      assert(
+        positions.includes(position),
```

Comment:
How would this work with WEIGHT? Seems like it only works if the chosenPositions is a list of strings. But types would help find cases like this.